newsroom/utils.py

```diff
@@ -666,11 +666,10 @@ def parse_objectid(value: Union[str, ObjectId]) -> Union[str, ObjectId]:
         return value
 
 
-def get_groups(company: Optional[Company], groups: list) -> list:
+def get_groups(company: Optional[Company], groups: List[Group]) -> List[Group]:
     """
-    filter agenda groups based on restrict_coverage_info
-    # STTNHUB - 327
+    filter agenda groups based on restrict_coverage_info permission
     """
     if company and company.get("restrict_coverage_info"):
-        return [item for item in groups if item["field"] != "stturgency"]
+        return [group for group in groups if "restrict_coverage_info" not in group.get("permissions", [])]
```

Comment:
it would be better to do the logic like:
```py
def get_groups(...):
    return [has_permissions(group, company) for group ...]

def has_permissions(group, company):
    group_permissions = group.get("permissions", [])
    if group_permissions:
        for permission in group_permissions:
            if not company or not company.get(permission):
                return False
        return True
    return True
```